### PR TITLE
feat: Support java21

### PR DESCRIPTION
**Pull request by astuyve**
### What does this PR do?

1. Support Java21

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: v2/.projen/deps.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "dependencies": [
     {
       "name": "@aws-cdk/aws-lambda-python-alpha",
-      "version": "2.113.0-alpha.0",
+      "version": "^2.101.1-alpha.0",
```
[the remark this answers is not in the document]
Comment (astuyve) [reply v2/.projen/deps.json:5]: hmm this is autogenerated

[the remark this answers is not in the document]
Comment (astuyve) [reply v2/.projen/deps.json:5]: Looks like we bumped this in package.json but not in projen? https://github.com/DataDog/datadog-cdk-constructs/blob/main/v2/.projenrc.js#L33

we removed the dependency from the project entirely, should this just be removed?

[the remark this answers is not in the document]
Comment (duncanista) [reply v2/.projen/deps.json:5]: Could have been my bad, if you approve #237, this should update it I think!

File: package.json [hunk 2]
```diff
@@ -6,4 +6,3 @@
         "loglevel"
     ]
 }
-
```
Comment (duncanista) [inline package.json:9]: nit: should we add this EOL?

File: v2/scripts/run_integration_tests.sh [hunk 3]
```diff
@@ -64,7 +64,7 @@ printOutputAndExit() {
 }
 
 for ((i = 0; i < ${#STACK_CONFIGS[@]}; i++)); do
-    tsc --project tsconfig.json
+    npx tsc --project tsconfig.json
```
Comment (duncanista) [inline v2/scripts/run_integration_tests.sh:67]: Was this change needed in your side?

Comment (astuyve) [reply v2/scripts/run_integration_tests.sh:67]: yes we shouldn't assume tsc is installed globally
